libyul/backends/wasm/EVMToEWasmTranslator.cpp

```diff
@@ -341,22 +771,50 @@ function u256_to_i32(x1, x2, x3, x4) -> v {
 	v := x4
 }
 
+function u256_to_byte(x1, x2, x3, x4) -> v {
+	if i64.ne(0, i64.or(i64.or(x1, x2), x3)) { invalid() }
+	if i64.gt_u(x4, 255) { invalid() }
+	v := x4
+}
+
 function u256_to_i32ptr(x1, x2, x3, x4) -> v {
 	v := u256_to_i32(x1, x2, x3, x4)
 }
 
+function u256_to_address(x1, x2, x3, x4) -> r1, r2, r3 {
+	if i64.ne(0, x1) { invalid() }
+	if i64.ne(0, i64.shr_u(x2, 4)) { invalid() }
+	r1 := x2
+	r2 := x3
+	r3 := x4
+}
+
 function keccak256(x1, x2, x3, x4, y1, y2, y3, y4) -> z1, z2, z3, z4 {
-	// TODO implement
-	unreachable()
+	mstore_internal(12, 0, 0, 0, 0x09)
+	z4 := eth.callStatic(
+		eth.getGasLeft(),
```

Comment:
It might be better to use a big constant instead, like `0x7fffffffffffffff`

Reply:
Done.